### PR TITLE
Add functionality to pass first 3 tests

### PR DESCRIPTION
**Pull request by smarquez-jpg**

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,23 @@
...
+import java.lang.*;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if(events.equals(Collections.emptySet())) return Arrays.asList(TimeRange.WHOLE_DAY);
```
Comment (peizhaoou) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: It seems a convention (at least for C++ and a glimpse on some java code as well) that always use {} to embrace the body of if/else/for, etc., even if there is only one statement inside it. That way, the scope is more clear, and can better avoid accidental dangle if/elseif, etc.

Comment (peizhaoou) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:25]: To check whether a Collection type is empty, use "events.isEmpty()".

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,23 @@
...
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) return Arrays.asList();
```
Comment (peizhaoou) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:26]: Same applies here, use {}

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,23 @@
...
+    Collection<TimeRange> options = new ArrayList<>();
+    for(Event event : events){
+      TimeRange beforeEvent = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, event.getWhen().start(), false);
```
Comment (peizhaoou) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:29]: If beforeEvent & afterEvent are only used once and immediately after, you could consider eliminating the variable altogether.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -31,6 +31,16 @@
...
+    Collection<TimeRange> updatedOptions = new ArrayList<>();
+    for(TimeRange firstTime : options){
+      for(TimeRange secondTime : options){
+        if(firstTime.overlaps(secondTime) && !(firstTime.equals(secondTime))){
+          TimeRange newOption = TimeRange.fromStartEnd(Math.max(firstTime.start(), secondTime.start()), Math.min(firstTime.end(), secondTime.end()), false);
+          if(!updatedOptions.contains(newOption)) updatedOptions.add(newOption);
+        }
+      }
+    }
+    if(updatedOptions.size() == 0) return options;
```
Comment (peizhaoou) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:43]: Nit: use {} as well.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -31,6 +31,16 @@
       TimeRange afterEvent = TimeRange.fromStartEnd(event.getWhen().end(), TimeRange.END_OF_DAY, true);
       options.add(afterEvent);
     }
-    return options;
+    Collection<TimeRange> updatedOptions = new ArrayList<>();
+    for(TimeRange firstTime : options){
+      for(TimeRange secondTime : options){
+        if(firstTime.overlaps(secondTime) && !(firstTime.equals(secondTime))){
```
Comment (peizhaoou) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:37]: Nit: the parentheses around the equality check is redundant, just use !firstTime.equals(secondTime)

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 6]
```diff
@@ -14,10 +14,33 @@
...
+    for(Event event : events){
+      TimeRange beforeEvent = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, event.getWhen().start(), false);
+      options.add(beforeEvent);
+      TimeRange afterEvent = TimeRange.fromStartEnd(event.getWhen().end(), TimeRange.END_OF_DAY, true);
+      options.add(afterEvent);
+    }
+    Collection<TimeRange> updatedOptions = new ArrayList<>();
+    for(TimeRange firstTime : options){
+      for(TimeRange secondTime : options){
+        if(firstTime.overlaps(secondTime) && !(firstTime.equals(secondTime))){
```
Comment (martiKallas) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:37]: nit: you should have a space after for and if and before the parenthesis: for (
you should have a space before almost any curly brace: if (...) { 

Please fix these throughout the file.

See: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace

**Pull request merged**